tour_model_eval/first_second_round_evaluation.ipynb

```diff
@@ -0,0 +1,352 @@
+    "                idx_labels_track.append([ori_idx])\n",
+    "\n",
+    "        # get first round labels\n",
+    "        first_labels = []\n",
+    "        for b in range(len(bins)):\n",
+    "            for trip in bins[b]:\n",
+    "                first_labels.append(b)\n",
+    "        new_labels = first_labels.copy()\n",
+    "        first_label_set = list(set(first_labels))\n",
+    "\n",
+    "        # store first round labels in idx_labels_track list\n",
+    "        for i in range(len(first_labels)):\n",
+    "            idx_labels_track[i].append(first_labels[i])\n",
+    "        # make a copy of idx_labels_track\n",
+    "        track = idx_labels_track.copy()\n",
+    "\n",
+    "        # get request percentage for the subset for the first round\n",
+    "        percentge_first = float('%.3f' % grp.get_req_pct(new_labels,track,filter_trips,sim))\n",
+    "        pct_collect_first_test.append(percentge_first)\n",
+    "\n",
+    "        # get homogeneity score for the subset for the first round\n",
+    "        homo_first = float('%.3f' % gs.score(bin_trips, new_labels))\n",
+    "        homo_collect_first_test.append(homo_first)\n",
+    "\n",
+    "        low = coll_low[k]\n",
+    "        dist_pct = coll_dist_pct[k]\n",
+    "\n",
+    "        # get second round labels\n",
+    "        for l in first_label_set:\n",
+    "            # store second round trips data\n",
+    "            second_round_trips = []\n",
+    "            # create a track to store indices and labels for the second round\n",
+    "            second_round_idx_labels = []\n",
+    "            for index, first_label in enumerate(first_labels):\n",
+    "                if first_label == l:\n",
+    "                    second_round_trips.append(bin_trips[index])\n",
+    "                    second_round_idx_labels.append([index, first_label])\n",
+    "            points = []\n",
+    "            point_features = []\n",
+    "\n",
+    "            for trip in second_round_trips:\n",
+    "                start = trip.data.start_loc[\"coordinates\"]\n",
+    "                end = trip.data.end_loc[\"coordinates\"]\n",
+    "                #                 hour = trip.data.start_local_dt['hour']\n",
+    "                distance = trip.data.distance\n",
+    "                duration = trip.data.duration\n",
+    "                points.append([start[0], start[1], end[0], end[1]])\n",
+    "                point_features.append([start[0], start[1], end[0], end[1], distance, duration])\n",
+    "\n",
+    "            x = np.array(point_features)\n",
+    "\n",
+    "            method = 'single'\n",
+    "            # get labels after two rounds of clustering on common trips\n",
+    "            new_labels = lp.get_new_labels(x, low, dist_pct, second_round_idx_labels, new_labels, method=method)           \n",
+    "            track = lp.change_track_labels(track, new_labels)\n",
+    "        \n",
+    "\n",
```

Comment:
this is essentially the same as the tuning set. Why are you copy-pasting the code?